src/engine/cachingreader/cachingreaderworker.cpp

```diff
@@ -114,27 +118,31 @@ void CachingReaderWorker::run() {
     }
 }
 
-void CachingReaderWorker::loadTrack(const TrackPointer& pTrack) {
+void CachingReaderWorker::cleanUpOldTrack() {
+    m_pAudioSource.reset(); // Close open file handles of the old track.
+
     // Discard all pending read requests
     CachingReaderChunkReadRequest request;
     while (m_pChunkReadRequestFIFO->read(&request, 1) == 1) {
         const auto update = ReaderStatusUpdate::readDiscarded(request.chunk);
         m_pReaderStatusFIFO->writeBlocking(&update, 1);
     }
+}
 
-    // Unload the track
-    m_pAudioSource.reset(); // Close open file handles
+void CachingReaderWorker::ejectTrack() {
+    // This is called with the engine stopped.
+    cleanUpOldTrack();
 
-    if (!pTrack) {
-        // If no new track is available then we are done
-        const auto update = ReaderStatusUpdate::trackUnloaded();
-        m_pReaderStatusFIFO->writeBlocking(&update, 1);
-        return;
-    }
+    const auto update = ReaderStatusUpdate::trackUnloaded();
+    m_pReaderStatusFIFO->writeBlocking(&update, 1);
+}
 
+void CachingReaderWorker::loadTrack(const TrackPointer& pTrack) {
     // Emit that a new track is loading, stops the current track
```

Comment:
The first part of the comment just repeats the following statement and is obvious, please delete it. Why do we need to mention what happens in the outside world? We should send a signal and don't need to care. Let's delete the whole comment.